tests/test_end_to_end.py

```diff
@@ -350,3 +350,53 @@ def test_builtin_plugins(tmp_path):
 def test_missing_object_exception():
     invalid_page_id = "11111111111111111111111111111111"
     assert run_n2y([invalid_page_id]) != 0
+
+
+def test_cache_effect(tmp_path):
```

Comment:
Great! I would finish this test and make sure it's running correctly (but is failing) using a stub `Cache` class.